src/dials/command_line/generate_mask.py

```diff
@@ -83,13 +89,31 @@ def generate_mask(
         experiments: An experiment list containing only one imageset.
         params: Masking parameters, having the structure defined in
             :data:`phil_scope`.
+        starting_masks: list of masks to combine together with the mask being
+            generated
 
     Returns:
         A list of masks, one for each imageset.
 
         A copy of :param:`experiments` with the masks applied (optional,
         only returned if :attr:`params.output.experiments` is set).
     """
+
+    if starting_masks:
+        starting_mask = list(starting_masks[0])
+        for mask in starting_masks[1:]:
+            for panel_idx in range(len(starting_mask)):
+                starting_mask[panel_idx] &= mask[panel_idx]
+        starting_mask = tuple(starting_mask)
+
+    # Check if only combining masks
+    if not experiments and starting_masks:
+        # Save the mask to file
+        log.info("Writing mask to %s", params.output.mask)
+        with open(params.output.mask, "wb") as fh:
+            pickle.dump(starting_mask, fh)
```

Comment:
Not for here but parenthetic comment: we really should move away from using `pickle` for masks